### PR TITLE
Release v2.14.0

### PR DESCRIPTION
**Pull request by zhiqiangjin**
### Notes:
- Supported Controller version: **UserConnect-5.4.1166** 
- Supported Terraform version: **v0.12.x**

### Features:
1. Implemented support for dynamically updating ``customized_route_advertisement`` in **aviatrix_aws_tgw_vpc_attachment**
2. Implemented support for SAML authentication for Controller login in **aviatrix_saml_endpoint**
3. New data source to support referencing specific private/public subnets
  - **data_source_aviatrix_vpc**
4. New resources to support AWS TGW inter-region peering
  - **aviatrix_aws_tgw_peering**
  - **aviatrix_aws_tgw_peering_domain_conn**
5. Implemented new resource to support connection to External Devices for Transit Network:
  - **aviatrix_transit_external_device_conn**

### Enhancements:
1. Added ``peering_ha_gw_name`` in **aviatrix_gateway**, and ``ha_gw_name`` in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** as computed values
2. Added ``peering_ha_private_ip`` in **data_source_aviatrix_gateway**, and ``ha_private_ip`` in **data_source_aviatrix_spoke_gateway** and **data_source_aviatrix_transit_gateway** as computed values

### Bug Fixes:
1. Fixed issue where OpenVPN configurations are unable to be modified when attached to a GeoVPN

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,27 @@
 ## 2.14.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-5.4.1166**
```
Comment (inc1t3Ful) [inline CHANGELOG.md:3]: we will see what version is out when it is out

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+4. New resources to support AWS TGW inter region peering
+  - **aviatrix_aws_tgw_peering**
+  - **aviatrix_aws_tgw_peering_domain_conn**
+5. New resources
+  - **aviatrix_transit_external_device_conn**
+
+### Enhancements:
+1. Added ``peering_ha_gw_name`` in **aviatrix_gateway**, and ``ha_gw_name`` in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** as computed value to output HA gateway name
+2. Added ``peering_ha_private_ip`` in **data_source_aviatrix_gateway**, and ``ha_private_ip`` in **data_source_aviatrix_spoke_gateway** and **data_source_aviatrix_transit_gateway** as computed value to output private IP of HA gateway
+3. Updated **data_source_aviatrix_gateway**, **data_source_aviatrix_spoke_gateway** and **data_source_aviatrix_transit_gateway** to be latest
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: This is very generic. What does this mean?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+
+### Bug Fixes:
+1. Fixed issue where OpenVPN is not able to be modified when attached to GeoVPN
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Fixed issue where OpenVPN configurations were unable to be modified when attached to a GeoVPN

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,27 @@
...
+2. Implemented support for controller login SAML Endpoint in **aviatrix_saml_endpoint**
+3. Implemented **data_source_aviatrix_vpc**
+4. New resources to support AWS TGW inter region peering
+  - **aviatrix_aws_tgw_peering**
+  - **aviatrix_aws_tgw_peering_domain_conn**
+5. New resources
+  - **aviatrix_transit_external_device_conn**
+
+### Enhancements:
+1. Added ``peering_ha_gw_name`` in **aviatrix_gateway**, and ``ha_gw_name`` in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway** as computed value to output HA gateway name
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Sounds redundant. Probably can change the ending to:

... as computed output values.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,27 @@
...
+2. Added ``peering_ha_private_ip`` in **data_source_aviatrix_gateway**, and ``ha_private_ip`` in **data_source_aviatrix_spoke_gateway** and **data_source_aviatrix_transit_gateway** as computed value to output private IP of HA gateway
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: *see above comment*

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,27 @@
 ## 2.14.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-5.4.1166** 
+- Supported Terraform version: **v0.12.x**
+
+### Features:
+1. Implemented support for dynamically updating ``customized_route_advertise`` in **aviatrix_aws_tgw_vpc_attachment**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for dynamically updating ``customized_route_advertisement`` in **aviatrix_aws_tgw_vpc_attachment**

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,27 @@
...
+2. Implemented support for controller login SAML Endpoint in **aviatrix_saml_endpoint**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for SAML authentication for Controller login in **aviatrix_saml_endpoint**

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,27 @@
...
+3. Implemented **data_source_aviatrix_vpc**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: To maintain consistent formatting following previous release notes:

3. New data source to support referencing specific private/public subnets:
  - **data_source_aviatrix_vpc**

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,27 @@
...
+4. New resources to support AWS TGW inter region peering
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: New resources to support AWS TGW inter-region peering:

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,4 +1,27 @@
...
+  - **aviatrix_aws_tgw_peering**
+  - **aviatrix_aws_tgw_peering_domain_conn**
+5. New resources
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented new resource to support connections to External Devices for Transit Network:

File: CHANGELOG.md (hunk 3 above)
Comment (inc1t3Ful) [reply CHANGELOG.md]: The workflow on how to update these configurations will be updated in the GeoVPN docs and VPN gateway docs

File: CHANGELOG.md (hunk 1 above)
Comment (zhiqiangjin) [reply CHANGELOG.md:3]: Will update it to latest version right before releasing.

**Pull request merged**